src/context_free.ml

```diff
@@ -196,60 +196,66 @@ module Generated_code_hook = struct
     | _ -> t.f context { loc with loc_start = loc.loc_end } x
 end
 
-let rec map_node_rec context ts super_call loc base_ctxt x =
-  let open Result in
```

Comment:
Might we simplify some of this code, and make it harder to accidentally drop some errors on the floor, if we made a `With_errors` module and use it in place of `Result`? Then we could use `return` and `>>=` and make most of the error handling implicit, rather than passing around errors explicitly.

Reply:
There are plenty of places where "errors" is used in multiple clauses, one of which has something else appended to it. It would be easy to forget to append "errors" to a new batch of errors, and not have it caught in any way.

Reply:
Actually, I see what you mean about this not working well for a fold. I think it's fine as-is.